doc/design/custom_xcode_schemes.md

```diff
@@ -0,0 +1,467 @@
+`FooTests.__internal__.__test_bundle` scheme contains configuration that builds `//Tests/FooTests`
+and executes `//Tests/FooTests` when testing is requested.
+
+## Introduction of Custom Xcode Schemes
+
+A new Starlark module called, `xcode_schemes`, provides functions for defining custom, Xcode
+schemes.
+
+### Defining a Simple Xcode Scheme
+
+Building on the previous example, let's define a scheme that combines the two targets,
+`//Sources/Foo` and `//Tests/FooTests`.
+
+```python
+load(
+    "@com_github_buildbuddy_io_rules_xcodeproj//xcodeproj:xcodeproj.bzl",
+    "xcode_schemes",
+    "xcodeproj",
+)
+
+_SCHEMES = [
+    xcode_schemes.scheme(
+        name = "Foo Module",
+        build_action = xcode_schemes.build_action(["//Sources/Foo"]),
+        test_action = xcode_schemes.test_action(["//Tests/FooTests"]),
+    ),
+]
+
+xcodeproj(
+    name = "generate_xcodeproj",
+    project_name = "Command Line",
+    tags = ["manual"],
+    schemes = _SCHEMES,
+)
+```
+
+The `xcode_schemes.scheme` function call defines a scheme with a user visible name of `Foo Module`.
+It is configured to build `//Sources/Foo` and `//Tests/FooTests`. (Targets listed in the
+test action and launch action are automatically added to the build action.) The scheme is also
+configured to execute the `//Tests/FooTests` target when testing is requested.
+
+The result of the function is wrapped in a `list` and passed to the `schemes` parameter of the
+`xcodeproj` macro.
+
+### Specifying a Launch Target
+
+Let's continue our example. We will add an `ios_application` and a `ios_ui_test` to the mix.
+
+```python
+# Assumptions
+#   //Sources/Foo:Foo - swift_library
+#   //Sources/FooTests:FooTests = ios_unit_test
+#   //Sources/App = ios_application
+#   //Sources/AppUITests = ios_ui_test
+
+load(
+    "@com_github_buildbuddy_io_rules_xcodeproj//xcodeproj:xcodeproj.bzl",
+    "xcode_schemes",
+    "xcodeproj",
+)
+
+_SCHEMES = [
+    xcode_schemes.scheme(
+        name = "Foo Module",
+        build_action = xcode_schemes.build_action(["//Sources/Foo"]),
+        test_action = xcode_schemes.test_action(["//Tests/FooTests"]),
+    ),
+    xcode_schemes.scheme(
+        name = "My Application",
+        test_action = xcode_schemes.test_action([
+            "//Tests/AppUITests",
+            "//Tests/FooTests",
+        ]),
+        launch_action = xcode_schemes.launch_action("//Sources/App"),
+    ),
+]
+
+xcodeproj(
+    name = "generate_xcodeproj",
+    project_name = "Command Line",
+    tags = ["manual"],
+    schemes = _SCHEMES,
+)
+```
+
+The above example adds a second scheme called `My Application`. It is configured with two test targets
+and a single launch target. All three targets are implicitly included in the build list.
+
+### Specifying Launch Arguments and Environment Variables
+
+There are times when it is desirabe to customize the launch action with arguments and environment
```

Comment:
Another launch configuration option that I use often is to specify a custom working directory. Is this something we could support configuring here?

Reply:
I think so, yes.

Reply:
Sure. Do you anticipate hardcoding an absolute path in the BUILD file or would you be using `$(location)`?

Reply:
What would `$(location)` resolve to? When I'm building CLIs I often want to run them in the project's root directory or an absolute directory.

Reply:
Digging into `tuist/XcodeProj`, I see [useCustomWorkingDirectory](https://github.com/tuist/XcodeProj/blob/3a93b47a34860a4d7dbcd9cc0ae8e9543c179c61/Sources/XcodeProj/Scheme/XCScheme%2BLaunchAction.swift#L45) that specifies that a custom working directory can be used. However, I don't see [customWorkingDirectory in the data model](https://github.com/tuist/XcodeProj/search?q=customWorkingDirectory). I confirmed that is the value that is written when you specify a custom working directory in Xcode. 

It looks like we will need to put up a PR to tuist to add it.

Reply:
I have updated the doc with `working_directory` support for `launch_action`.